### PR TITLE
Introduce CSV related tools to master

### PR DESCRIPTION
**Pull request by karolk91**
Add a CSV creation script that parses smart-bench logs and converts them into entries in CSV format

### CODE REVIEW

File: stats/README.md [hunk 1]
```diff
@@ -5,9 +5,23 @@ This folder contains utilities to create visual graphs based on smart-bench meas
...
+        cargo run --release -- evm flipper --instance-count 1 --call-count 1500 --url ws://localhost:9988 | ./smart_bench_to_csv.sh --csv-output=benchmark-result.csv --timestamp=1714515934
+
+        cargo run --release -- ink-wasm flipper --instance-count 1 --call-count 1500 --url ws://localhost:9988 | ./smart_bench_to_csv.sh --csv-output=benchmark-result.csv --timestamp=1714515934
+
+        cargo run --release -- sol-wasm flipper --instance-count 1 --call-count 1500 --url ws://localhost:9988 | ./smart_bench_to_csv.sh --csv-output=benchmark-result.csv --timestamp=1714515934
+        ```
+        above will create `benchmark-result.csv` file with all `3` results appended
+    
+    1. or get existing csv results from [gh-pages branch](https://github.com/
+    paritytech/smart-bench/blob/gh-pages/benchmark-results.csv)
```
Comment (smiasojed) [inline stats/README.md:20]:
```suggestion
    paritytech/smart-bench/blob/gh-pages/bench_<CONTRACT>.csv)
```

File: launch/README.md [hunk 2]
```diff
@@ -17,7 +17,7 @@ Please refer to the [download-bin.sh](./download-bin.sh) script for detailed inf
 It is possible to run `smart-bench:latest` using raw `docker run` command with options (see examples below) but usage of `run.sh` can abstract away some of the internals of the image and also result in better compatibility with future image versions. The `run.sh` currently helps with following usage scenarios:
 1. use `smart-bench:latest` image as is, composed of specific versions of nodes and contracts
 1. use `smart-bench:latest` for what it consists of but provide custom set of pre-compiled contracts
-1. use `smart-bench:latest` but provide custom set of node binaries (`polkadot`, `zombienet` `polkadot-parachain`, `moonbeam`)
+1. use `smart-bench:latest` but provide custom set of node binaries (`polkadot`, `zombienet` `polkadot-parachain`, `polkadot-execute-worker`, `polkadot-prepare-worker`, `moonbeam`)
```
Comment (smiasojed) [inline launch/README.md:20]:
```suggestion
1. use `smart-bench:latest` but provide custom set of node binaries (`polkadot`, `zombienet` `polkadot-parachain`, `moonbeam`)
```

File: launch/README.md [hunk 3]
```diff
@@ -130,17 +130,40 @@ docker run --rm -it --init -v $PWD/configs:/usr/local/smart-bench/config smart-b
 ---
 
 ## Miscellaneous
+
```
Comment (smiasojed) [inline launch/README.md:133]: We are not using prefund account on master - master is compatible with polkadot 1.0.0

File: launch/README.md [hunk 4]
```diff
@@ -130,17 +130,40 @@ docker run --rm -it --init -v $PWD/configs:/usr/local/smart-bench/config smart-b
...
+```
+
 ### Moonbeam with Dev RPC module enabled build recipe
 
 Following is an example recipe how to build moonbeam binary with Dev RPC module enabled
 ```
 git clone https://github.com/PureStake/moonbeam.git
 cd moonbeam && git fetch https://github.com/karolk91/moonbeam master && git cherry-pick decd877
-docker run -it --rm -v $PWD:/moonbeam docker.io/paritytech/ci-linux:production /bin/bash
+docker run -it --rm -v $PWD:/moonbeam docker.io/paritytech/ci-unified:bullseye-1.74.0 /bin/bash
```
Comment (smiasojed) [inline launch/README.md:162]: Please keep here old versions

**Comment by smiasojed** [comment]
I would like to keep master compatible with polkadot v1.0.0, once we merge the csv generation script I will move the tag v1.0.0 to head of master.
The version v1.7.0 we will keep on v1.7.0 branch

File: launch/README.md [hunk 5]
```diff
@@ -4,10 +4,10 @@ This directory contains docker related scripts and configurations to compile and
...
 
-At the first stage of docker build, based upon `docker.io/paritytech/ci-linux:production` as rust enabled compilation env, smart-bench software is being compiled from current sources (this git repository). Resulting binary is copied into second stage / final docker image based on `ubuntu:20.04` image where also other dependencies are configured and installed.
+At the first stage of docker build, based upon `docker.io/paritytech/ci-unified:bullseye-1.74.0` as rust enabled compilation env, smart-bench software is being compiled from current sources (this git repository). Resulting binary is copied into second stage / final docker image based on `ubuntu:20.04` image where also other dependencies are configured and installed.
 
 Final image consists of few pieces that are integrated into it: 
-- `polkadot`, `zombienet` and `polkadot-parachain` pre-compiled binaries taken from offical releases
+- `polkadot`, `polkadot-parachain`, `polkadot-execute-worker`, `polkadot-prepare-worker` and `zombienet` pre-compiled binaries taken from offical releases
```
Comment (smiasojed) [inline launch/README.md:10]:
```suggestion
- `polkadot`, `polkadot-parachain` and `zombienet` pre-compiled binaries taken from offical releases
```

File: launch/README.md [hunk 6]
```diff
@@ -4,10 +4,10 @@ This directory contains docker related scripts and configurations to compile and
...
 - `moonbeam` binary provided by custom release created on [this fork](https://github.com/karolk91/moonbeam) (custom fork was required to enable dev rpc module which is not supported by offical releases)
```
Comment (smiasojed) [inline launch/README.md:11]: which branch is compatible with polkadot v1.0.0?

File: stats/smart_bench_to_csv.sh [hunk 7]
```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
```
Comment (smiasojed) [inline stats/smart_bench_to_csv.sh:1]: maybe we could add output-csv option to smart-bench?

File: launch/README.md (hunk 3 above)
Comment (karolk91) [reply launch/README.md:133]: will remove the section, thanks for spotting this

File: stats/smart_bench_to_csv.sh (hunk 7 above)
Comment (karolk91) [reply stats/smart_bench_to_csv.sh:1]: sure, thats good idea, we can think of this pull request as some intermediate step towards better solution

File: launch/README.md (hunk 6 above)
Comment (karolk91) [reply launch/README.md:11]: its the same moonbeam version for both branches

File: stats/smart_bench_to_csv.sh (hunk 7 above)
Comment (smiasojed) [reply stats/smart_bench_to_csv.sh:1]: OK, lets create an issue for improvement

Comment (ascjones) [reply stats/smart_bench_to_csv.sh:1]: I agree this might be a better idea just to add the option to output as csv to smart-bench itself.

**Review by ascjones — APPROVED** [review]

**Pull request merged**